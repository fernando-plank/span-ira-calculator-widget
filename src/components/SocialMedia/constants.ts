export const socialMedias = [
  {
    icon: 'Instagram',
    url: 'https://www.instagram.com/get_span/?hl=en'
  },
  {
    icon: 'Linkedin',
    url: 'https://www.linkedin.com/company/spanio/mycompany/'
  },
  {
    icon: 'Twitter',
    url: 'https://twitter.com/span_io?lang=en'
  },
  {
    icon: 'Youtube',
    url: 'https://www.youtube.com/channel/UClLe-YG5QZKFJAlJhwdYPGA'
  },
  {
    icon: 'Facebook',
    url: 'https://www.facebook.com/get.SPAN/'
  },
  {
    icon: 'Tiktok',
    url: 'https://www.tiktok.com/@span.io?lang=en'
  }
]
