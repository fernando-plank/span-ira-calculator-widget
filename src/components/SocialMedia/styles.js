import styled from 'styled-components';
export const Link = styled.a `
  margin-bottom: 15px;
`;
