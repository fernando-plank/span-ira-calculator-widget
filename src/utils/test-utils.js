var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
/* eslint-disable import/export */
import React from 'react';
import theme from '@styles/theme';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
jest.mock('../components/Icon', () => {
    return jest.fn(() => React.createElement("div", null, "mocked icon"));
});
const customRender = (ui, _a = {}) => {
    var renderOptions = __rest(_a, []);
    return render(React.createElement(ThemeProvider, { theme: theme }, ui), renderOptions);
};
export * from '@testing-library/react';
export { customRender as render };
