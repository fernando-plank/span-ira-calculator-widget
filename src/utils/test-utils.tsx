/* eslint-disable import/export */
import { ReactElement } from 'react'
import { RenderOptions, render } from '@testing-library/react'

import theme from '@styles/theme'
import { ThemeProvider } from 'styled-components'

jest.mock('../components/Icon', () => {
  return jest.fn(() => <div>mocked icon</div>)
})

const customRender = (ui: ReactElement, { ...options }: RenderOptions = {}) =>
  render(<ThemeProvider theme={theme}>{ui}</ThemeProvider>, options)

export * from '@testing-library/react'
export { customRender as render }
