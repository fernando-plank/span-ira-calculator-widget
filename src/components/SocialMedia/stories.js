import React from 'react';
import SocialMedia from '.';
export default {
    title: 'SocialMedia',
    component: SocialMedia
};
export const Default = () => React.createElement(SocialMedia, { icon: "Facebook", url: "#" });
