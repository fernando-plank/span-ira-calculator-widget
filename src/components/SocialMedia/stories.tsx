import React from 'react'
import { Story, Meta } from '@storybook/react'

import SocialMedia from '.'

export default {
  title: 'SocialMedia',
  component: SocialMedia
} as Meta

export const Default: Story = () => <SocialMedia icon="Facebook" url="#" />
