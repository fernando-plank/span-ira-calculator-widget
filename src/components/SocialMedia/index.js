import React from 'react';
import Icon from '@components/Icon';
import * as S from './styles';
const SocialMedia = (props) => (React.createElement(S.Link, { "data-testid": "social-media-link", target: "_blank", href: props.url },
    React.createElement(Icon, { icon: props.icon })));
export default SocialMedia;
