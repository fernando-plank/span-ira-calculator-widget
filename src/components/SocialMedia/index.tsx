import React from 'react'
import Icon from '@components/Icon'

import * as S from './styles'

export interface SocialMediaProps {
  icon: string
  url: string
}

const SocialMedia = (props: SocialMediaProps) => (
  <S.Link data-testid="social-media-link" target="_blank" href={props.url}>
    <Icon icon={props.icon} />
  </S.Link>
)

export default SocialMedia
