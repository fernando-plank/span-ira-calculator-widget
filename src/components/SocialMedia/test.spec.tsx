import { screen } from '@testing-library/react'
import { render } from '@utils/test-utils'

import SocialMedia from '.'

jest.mock('../Icon', () => {
  return jest.fn(() => <div>mocked icon</div>)
})

describe('<SocialMedia />', () => {
  it('should render social media link', () => {
    const { container } = render(
      <SocialMedia icon={'Facebook'} url={'http://facebook.com'} />
    )

    expect(screen.getByTestId('social-media-link')).toHaveProperty(
      'href',
      'http://facebook.com/'
    )
    expect(container.firstChild).toMatchSnapshot()
  })
})
