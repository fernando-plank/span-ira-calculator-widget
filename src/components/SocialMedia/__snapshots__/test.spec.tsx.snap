// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SocialMedia /> should render social media link 1`] = `
.c0 {
  margin-bottom: 15px;
}

<a
  class="c0"
  data-testid="social-media-link"
  href="http://facebook.com"
  target="_blank"
>
  <div>
    mocked icon
  </div>
</a>
`;
